@kiva/kv-shop/src/basketCredits.ts

```diff
@@ -54,9 +54,9 @@ export interface ApplyPromoCreditData {
 export async function applyPromoCredit(
 	apollo: ApolloClient<any>,
 	options: MutationOptions<any>,
-): Promise<boolean> {
-	if (!options?.variables?.creditType && !options?.variables?.redemptionCode) {
-		return Promise.resolve(false);
+): Promise<any> {
+	if (!options?.variables?.creditType || !options?.variables?.redemptionCode) {
```

Comment:
This is not an either or scenario. Both must be present or the mutation will fail.

Reply:
Don't we want the function to return false whether the credit type OR the redemption code is missing?
This is a negation so || seems like the right operator, not &&.  && would be if we weren't negating.

That is, if both must be present for the mutation not to fail...

Reply:
Otherwise we'd be requiring that both be missing in order for the mutation to return false.

Reply:
Yea I guess it will work either way. If either of these are missing we'll return...

Reply:
Good catch on this!